pkg/runner/function_scheduler/function_scheduler.go

```diff
@@ -10,10 +13,12 @@ import (
 )
 
 var (
+	once = &sync.Once{}
 	fs *FunctionScheduler
 )
 
 func GetFunctionScheduler() *FunctionScheduler {
+	once.Do(FunctionSchedulerInit)
```

Comment:
reasonable and impressive

Reply:
The `Once` reminds me that we can use `Once` to easily implement a `Reset` action, and it can help you to deal with the singleton test.

For example, you use `Once` to init a new instance, let's say `p *sync.Once`, and when you want to `Reset`, just offer p a new `Once` instance `p = new(sync.Once)`

Reply:
but the `Once` it is still a package level static variable. Which means I reset the `Once` instead. Now I just create a new singleton instance to test.